mlflow/java/client/src/main/java/org/mlflow/tracking/Page.java

```diff
@@ -0,0 +1,35 @@
+package org.mlflow.tracking;
+
+import java.util.List;
+import java.util.Optional;
+
+interface Page<E> {
+
+  /**
+   * @return Returns the number of elements in this page.
+   */
+  public int getPageSize();
+
+  /**
+   * @return Returns true if there are more pages that can be retrieved from the API.
+   */
+  public boolean hasNextPage();
+
+  /**
+   * @return Returns an Optional of the token string to get the next page. 
+   * Empty if there is no next page.
+   */
+  public Optional<String> getNextPageToken();
+
+  /**
+   * @return Retrieves the next Page object using the next page token,
+   * or returns an empty page if there are no more pages.
+   */
+  public Page<E> getNextPage();
+
+  /**
+   * @return Returns a List of the elements in this Page.
+   */
+  public List<E> getItems();
```

Comment:
nit: I think we should actually just return an `iterable` here. This way we don't make any assumptions about how the Page interface is implemented. A super contrived example of why you might want to return an `Iterable` is if you wanted to implement a page which somehow streams it's contents instead of materializing them all into memory...